.github/workflows/rust.yml

```diff
@@ -24,6 +24,11 @@ jobs:
           ~/.sp1/bin/sp1up --token ${{ secrets.GITHUB_TOKEN }}
           ~/.sp1/bin/cargo-prove prove --version
 
+      - name: Install just
+        uses: extractions/setup-just@v2
+      - name: Build SP1 Programs
+        run: just build-sp1-programs
```

Comment:
Just so I understand, this is added for things to compile when you run fmt/clippy?

Reply:
Would it be better if we used just recipes instead for the listing so that any pre-building can be done in the recipe? If you agree, we can create a separate issue for it.

Reply:
I mean the lint commands below here are done directly, but we have them as just commands/recipes. Those can have dependencies to make sure whatever is needed to be built is built?

Reply:
Let's not create an issue for this. I don't like to rely on `just` when it comes to the main function of a workflow. I actually want to test the command I'm running, not the recipe. I view the `just build-sp1-programs` as a dependency to the actual command I wanna test.

Notice that when I'm testing whether sp1 programs build, I don't actually run `just`. (Well I noticed I'm missing those workflows, although they are indirectly tested, I probably want to add them back)